client/client.go

```diff
@@ -76,52 +71,52 @@ func (c *LogClient) AddLeaf(ctx context.Context, data []byte) error {
 	switch {
 	case grpc.Code(err) == codes.AlreadyExists:
 		// If the leaf already exists, don't wait for an update.
-		return c.getInclusionProof(ctx, leaf.MerkleLeafHash, c.str.TreeSize)
+		return c.getInclusionProof(ctx, leaf.MerkleLeafHash, c.root.TreeSize)
 	case err != nil:
 		return err
 	default:
 		err := grpc.Errorf(codes.NotFound, "Pre-loop condition")
-		for i := 0; grpc.Code(err) == codes.NotFound && i < c.maxTries; i++ {
+		for i := 0; grpc.Code(err) == codes.NotFound && ctx.Err() == nil; i++ {
 			// Wait for TreeSize to update.
-			if err := c.waitForSTRUpdate(ctx, c.maxTries); err != nil {
+			if err := c.waitForRootUpdate(ctx); err != nil {
 				return err
 			}
 
 			// Get proof by hash.
-			err = c.getInclusionProof(ctx, leaf.MerkleLeafHash, c.str.TreeSize)
+			err = c.getInclusionProof(ctx, leaf.MerkleLeafHash, c.root.TreeSize)
 		}
 		return err
 	}
 }
 
-// waitForSTRUpdate repeatedly fetches the STR until the TreeSize changes
-// or until a maximum number of attempts has been tried.
-func (c *LogClient) waitForSTRUpdate(ctx context.Context, attempts int) error {
+// waitForRootUpdate repeatedly fetches the Root until the TreeSize changes
+// or until ctx times out.
+func (c *LogClient) waitForRootUpdate(ctx context.Context) error {
 	b := &backoff.Backoff{
 		Min:    100 * time.Millisecond,
 		Max:    10 * time.Second,
 		Factor: 2,
 		Jitter: true,
 	}
-	startTreeSize := c.str.TreeSize
+	startTreeSize := c.root.TreeSize
 	for i := 0; ; i++ {
-		if err := c.UpdateSTR(ctx); err != nil {
+		if err := c.UpdateRoot(ctx); err != nil {
 			return err
 		}
-		if c.str.TreeSize > startTreeSize {
+		if c.root.TreeSize > startTreeSize {
 			return nil
 		}
-		if i >= (attempts - 1) {
+		if err := ctx.Err(); err != nil {
 			return grpc.Errorf(codes.DeadlineExceeded,
-				"UpdateSTR().TreeSize: %v, want > %v. Tried %v times.",
-				c.str.TreeSize, startTreeSize, i+1)
+				"%v. TreeSize: %v, want > %v. Tried %v times.",
+				err, c.root.TreeSize, startTreeSize, i+1)
 		}
 		time.Sleep(b.Duration())
 	}
 }
 
-// UpdateSTR retrieves the current SignedLogRoot and verifies it.
-func (c *LogClient) UpdateSTR(ctx context.Context) error {
+// UpdateRoot retrieves the current SignedLogRoot and verifies it.
```

Comment:
Maybe explain what "verifies" means either here or in the definition of the verifying client.